.github/workflows/ci.yml

```diff
@@ -0,0 +1,83 @@
+name: ci
+
+on: [push, pull_request]
```

Comment:
Might be a good idea to restrict these to run on `master` only. If someone wants to contribute using a feature branch, then the CI won't be redundantly run on both the pushes and the PR.

Reply:
Like [so](https://github.com/Dobiasd/enterprython/commit/4cda12f4d735105c5ef8da8eceb00a00a913ae19)? :slightly_smiling_face:

Reply:
Yes, exactly.

Reply:
Sounds good, I'll update the PR.

Reply:
Doing this will stop CI from running on feature branches, but PRs targeting the `master` branch will still run as they should.

Not having this filter will run the CI both on the contributor's repository and this repository. Kind of wasteful from an environmental point of view.

Reply:
But contributors (like me) will get feedback only when opening a PR. Typically PRs are in a mergeable state (or at least close to it) when submitted and don't need many builds within this repository. Removing the feedback for all developers makes a PR the only way to get basic quality checks.

From a developer point of view the quality development and software outweigh a few additional builds. Especially if the builds are within minutes anyway.

Reply:
Oh, right. That's also not good. I think I'd [prefer](https://github.com/Dobiasd/enterprython/commit/28d57a2f5656933f304669343dea6acfcc960967) redundant builds over missing builds. It's a bit weird to me that there is no canonical solution to get the best of both worlds.

Reply:
Adding `workflow_dispatch` to `on` will enable manual runs. I'd say that is the best of both worlds.

Reply:
> I'd say that is the best of both worlds.

To be honest: No. Why should one explicitely want to trigger each build manually?

Developing this PR I would have pressed _build_ more than 100 times (literally).

Reply:
Since I have both of your 👍 I'm going to resolve here 😄 . Thanks for your feedback!